src/apps/tempus/contracts/index.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable, Inject } from '@nestjs/common';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { ContractFactory } from '~contract/contracts';
+import { Network } from '~types/network.interface';
+
+import { Amm__factory } from './ethers';
+import { Pool__factory } from './ethers';
+import { PyToken__factory } from './ethers';
+
+// eslint-disable-next-line
+type ContractOpts = { address: string; network: Network };
+
+@Injectable()
+export class TempusContractFactory extends ContractFactory {
+  constructor(@Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit) {
+    super((network: Network) => appToolkit.getNetworkProvider(network));
+  }
+
+  amm({ address, network }: ContractOpts) {
+    return Amm__factory.connect(address, this.appToolkit.getNetworkProvider(network));
+  }
+  pool({ address, network }: ContractOpts) {
+    return Pool__factory.connect(address, this.appToolkit.getNetworkProvider(network));
+  }
+  pyToken({ address, network }: ContractOpts) {
```

Comment:
Ideally, prefix the JSON files with `tempus` so that the contract interfaces are importable elsewhere in the application as `TempusAmm`, `TempusPool`, etc.

It helps a little with autocomplete features.

(Not a blocking comment, just a suggestion that we'll add to the Studio docs later!)